### DISCUSSION TITLE
How to statically generate a page if data exists during build time and render on the server side if it doesnt?

### DISCUSSION THREAD

**Discussion by lkenny97** [vercel/next.js#34422] [Help]
I have a website that apart from main features has a blog section. I used to generate my blog pages statically since I did not have lots of them and I could afford to stop the server to add new ones before rebuilding the website again. Now my website has more users and I do not want to turn it off just to add some blogs. However, I still want to build my existing blog pages statically and use a CDN for css and js.

Is there a way to have one [slug].ts and build it statically for all the blogs existing in my db and render it on the server for blogs that I will add in the future until my next build?

I have not used getStaticProps and getServerSideProps together yet but I feel like that will not work.

**Comment by iankduffy** [comment]
You can use Fallback https://nextjs.org/docs/api-reference/data-fetching/get-static-paths#fallback-blocking

**Comment by 2stash** [comment]
Can't you just have your getStaticProps and getStaticPaths dynamically generate the pages? Meaning you use an api call that returns an array of posts, and if the length of the array changes, then you render more pages in getStaticProps and getStaticPaths. In the fallback example the pre-rendered pages are hard coded, which I don't think is necessary.  

edit: Setting fallback to true will let any page not prerendered be rendered in your [...slug].js page. I guess my point was I assume you don't have the pages you want prerendered typed out in your getStaticPaths. If you do then just change fallback to true and you shouldn't have to do anything else. If you don't have your paths hardcoded, you might have to many pages to prerender them all, so you might want to still set fallback to true, and maybe only prerender some, maybe in your data fetch you add some logic to only fetch and prerender the most popular that are probably linked on the front of the blog.

For your example, I don't think you need to use getServerSideProps since the data won't change much and you can just use getStaticProps for the blog, and also set a revalidate prop if needed.
